cameraview/src/main/java/com/otaliastudios/cameraview/CameraView.java

```diff
@@ -77,6 +81,10 @@
     private Handler mUiHandler;
     private WorkerHandler mFrameProcessorsHandler;
 
+    // Overlay
+    private boolean mHasOverlay = false;
+    private OverlayLayout mPreviewOverlayLayout;
```

Comment:
Can you instantiate the OverlayLayout in this class and add it with addView, just like we do with the gesture layout for example?

I think that `CameraPreview`s should know nothing about overlays. We will draw over the preview using android View system.